app/hotels/src/allHotels/AllHotelsSearchList.js

```diff
@@ -31,6 +32,8 @@ function AllHotelsSearchList(combinedProps: Props) {
         }
       })}
     </ScrollView>
+  ) : (
+    <GeneralError errorMessage="Server error, please try again later" />
```

Comment:
You sure? Is it server error? What if there are no edges? This is the most problematic part and one of the reasons why I don't know how to fix it properly...

What about instead of writing "server error" something like "it was not possible to load/find" or "no hotels found".

Reply:
I would expect that the API returns an empty array if no hotel is found and `!![] === true`.

Reply:
Ok, this is even worse then. If it returns the empty array the result will be empty screen without the error message, right?

Reply:
You're right, we need to handle both.

Reply:
Try to think about reusability please, because there are so many places where we need it. Maybe something like "error boundaries" component? (inspired by React itself but with completely different functionality)...